components/ledger/internal/adapters/http/in/transaction_create.go

```
@@ -798,128 +636,190 @@ func (handler *TransactionHandler) buildStandardOp(
+func (handler *TransactionHandler) createTransaction(c *fiber.Ctx, transactionInput pkgTransaction.Transaction, transactionStatus string) error {
+	return handler.executeCreateTransaction(c, transactionInput, transactionStatus, false)
+}
+
+// createRevertTransaction creates a reversal transaction. The action is forced
+// to "revert" so that accounting route lookups use the revert rubrics instead
+// of the status-derived action.
+func (handler *TransactionHandler) createRevertTransaction(c *fiber.Ctx, transactionInput pkgTransaction.Transaction, transactionStatus string) error {
+	return handler.executeCreateTransaction(c, transactionInput, transactionStatus, true)
+}
+
+func (handler *TransactionHandler) executeCreateTransaction(c *fiber.Ctx, transactionInput pkgTransaction.Transaction, transactionStatus string, isRevert bool) error {
```

golangci-lint (2.11.4)

[error] cyclomatic complexity 21 of func `(*TransactionHandler).executeCreateTransaction` is high (> 18)

(gocyclo)